### ISSUE TITLE
SC2120 $@ false positive

### ISSUE THREAD

**Issue by masavini** [koalaman/shellcheck#1115]
```sh
function __wiki_raw_link_remover () {
  sed \
    -e 's/\[\[[^|]\+|\([^]]\+\)/[[\1/g' \
    -e 's/\[\[//g' \
    -e 's/\]\]//g' \
    -- "${@}"
}

cat wiki-raw-file.txt | __wiki_raw_link_remover
```
running shellcheck on the above script results in:
```
In wiki_helpers.sh line 1:
__wiki_raw_link_remover() {
^-- SC2120: __wiki_raw_link_remover references arguments, but none are ever passed.
```
`${@}` as argument for sed, grep, awk (and friends) is great, as it allows to write wrapper functions that can process both input file(s) or standard input.

**Comment by koalaman** [comment]
You're right of course. Unfortunately I don't see any good way of detecting that this was intentional. Someone could equally well have written similar code expecting `"$@"` to be the script's arguments.
